Documentation added for Rows class.

Shershon25 opened the pull request:
Have added documentation for the Rows class. Suggestions and corrections are most welcomed.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5719,12 +5719,48 @@ def asdict(self):
+        args:
+        table- accepts a table object of class Table()
+
+        >>>  t = td.Table().with_columns({
```
I'm unsure why we're not instantiating the `Table` class directly here. What's the `td` object here? I don't see it anywhere in the docs?

Shershon25 replied:
That was a mistake on my side. I didn't add the proper initialization for Table() class. I have updated it now.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5719,12 +5719,48 @@ def asdict(self):
+                'points': [  1,   2,   2,  10],
+            })
+
+            >>> td.Table().Rows(t).__getitem__(0)
```
You can reuse objects from previous docs. Let's save one Rows object in the class initializer and reuse that for the remaining methods in this class.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5735,9 +5771,35 @@ def __getitem__(self, i):
 
         def __len__(self):
+            """
+            Returns the no.of rows in the table.
```
please no short-forms in the docs, as much as possible

adnanhemani commented on datascience/tables.py:
```diff
@@ -5735,9 +5771,35 @@ def __getitem__(self, i):
+             'count':  [  9,   3,   3,   1],
+             'points': [  1,   2,   2,  10],
+            })
+            >>> td.Table().Rows(t).__len__()
```
We can jus use the Python `len` function to demonstrate this function's logic

Shershon25 replied:
Noted.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5719,12 +5719,48 @@ def asdict(self):
     class Rows(collections.abc.Sequence):
-        """An iterable view over the rows in a table."""
+        """
+        An iterable row-styled view of the table object that is passed as an argument.
```
Let's keep the original statement here - the intended/standard usage of this class doesn't have users manually creating this class.

adnanhemani reviewed (changes requested):
Thanks for this! Let's try to make the revisions here and then we should be in a good state to merge!

Shershon25 commented:
@adnanhemani I have updated the docs according to the changes suggested.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5678,12 +5678,46 @@ def asdict(self):
+        args:
+        table- accepts a table object of class Table()
+        
+        >>> from datascience import Table
```
`Table` should already be imported from other doctests above, no need to re-import it.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5678,12 +5678,46 @@ def asdict(self):
+        })
+
+        >>> r = Table.Rows(t)
+            print(r)
```
In doctests, we don't need to `print` any values. You can simply call `>>> r` and it will output the right representation of the object.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5678,12 +5678,46 @@ def asdict(self):
+        An iterable view over the rows in a table.
+
+        args:
+        table- accepts a table object of class Table()
```
Can we please follow the standard formatting for args/return values? Here's a good example: https://github.com/data-8/datascience/blob/master/datascience/tables.py#L380-L402

E.g.:
```
...

Args:
    table (str): ...
```

adnanhemani replied:
Also "Table" here can be written as Table or `Table` - but no parenthesis please.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5678,12 +5678,46 @@ def asdict(self):
+            """
+            Access the i-th row of the given table.
+
+            args:
```
Format comment, same as above

adnanhemani commented on datascience/tables.py:
```diff
@@ -5678,12 +5678,46 @@ def asdict(self):
+            returns:
+            Returns a Row object containing the i-th row of the given table
+
+            >>> r.__getitem__(0)
```
To show `__getitem__`, you can just use the `[]` notation. Example: `r[0]` should give you the same result. Change required because it is more Pythonic and easier to read/remember for beginning learners.

adnanhemani commented on datascience/tables.py:
```diff
@@ -5694,9 +5728,24 @@ def __getitem__(self, i):
+            Returns the printable representation of the given table as string.
+            Uses the standard repr() function.
+
+            >>> r.__repr__()
```
Similar as the last comment, but we should show this as `repr(r)` instead.

adnanhemani reviewed (changes requested):
Hi, getting closer! In addition to the comments, the doc build is now failing as well :/ (check the GH Action attached to the repo). Please fix these, thank you!

Shershon25 commented:
Hey! I have updated the docs again. I corrected the incorrect prefix because of which the build was failing. If any other corrections are there, do let me know.

adnanhemani commented:
Hi! I think this is awesome!! Style and comments-wise, I think I'm ready to approve. But the build is failing (see above). Can you please take a look and ensure that all tests pass? Once that's done, I think I can approve and merge! Thank you for your hard work on this!

adnanhemani commented:
Hi, your changes were still failing the build, so I went ahead and made the required changes to get it to work. Thank you for this work!!
